From a8579a150f7c5ab2ab2da263d0e0fecb3b56cf8c Mon Sep 17 00:00:00 2001
From: Bobby The Builder <bob@najdan.com>
Date: Mon, 29 Aug 2022 12:52:58 -0400
Subject: [PATCH] Revert "arm64: dts: rk3399-pinephone-pro: Add support for
 powering up the modem"

This reverts commit d64d21f84913daae6538bdcbc3e6f4a6dec80726.
---
 .../dts/rockchip/rk3399-pinephone-pro.dts     | 76 -------------------
 1 file changed, 76 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts b/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
index 615995f28eb5..b4f0bc2410c0 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
@@ -423,30 +423,6 @@ vcc3v3_sys: wifi_bat: vcc3v3-sys {
 		vin-supply = <&vcc_sysin>;
 	};
 
-	vcc_4g_5v: vcc-4g-5v {
-		compatible = "regulator-fixed";
-		enable-active-high;
-		gpio = <&gpio1 RK_PC7 GPIO_ACTIVE_HIGH>;
-		pinctrl-names = "default";
-		pinctrl-0 = <&vcc_4g_5v_en>;
-		regulator-name = "vcc_4g_5v";
-		regulator-min-microvolt = <5000000>;
-		regulator-max-microvolt = <5000000>;
-		vin-supply = <&vcc5v0_sys>;
-	};
-
-	vcc_4g: vcc-4g {
-		compatible = "regulator-fixed";
-		enable-active-high;
-		gpio = <&gpio4 RK_PC7 GPIO_ACTIVE_HIGH>;
-		pinctrl-names = "default";
-		pinctrl-0 = <&vcc_4g_en>;
-		regulator-name = "vcc_4g";
-		regulator-min-microvolt = <3800000>;
-		regulator-max-microvolt = <3800000>;
-		vin-supply = <&vcc_sysin>;
-	};
-
 	vcc1v8_codec: vcc1v8-codec-regulator {
 		compatible = "regulator-fixed";
 		enable-active-high;
@@ -1196,35 +1172,6 @@ mipi_in_panel: endpoint {
 	};
 };
 
-&uart3 {
-	status = "okay";
-
-	modem {
-		compatible = "quectel,eg25";
-		char-device-name = "modem-power";
-
-		pinctrl-names = "default";
-		pinctrl-0 = <&modem_control_pins>;
-
-		power-supply = <&vcc_4g>;
-		vbus-supply = <&vcc_4g_5v>;
-
-		enable-gpios = <&gpio0 RK_PB0 GPIO_ACTIVE_HIGH>; // W_DISABLE#
-		reset-gpios = <&gpio3 RK_PB0 GPIO_ACTIVE_HIGH>;
-		status-gpios = <&gpio3 RK_PA6 GPIO_ACTIVE_HIGH>;
-		pwrkey-gpios = <&gpio0 RK_PB5 GPIO_ACTIVE_HIGH>;
-
-		host-ready-gpios = <&gpio0 RK_PB4 GPIO_ACTIVE_HIGH>; // apready
-		wakeup-gpios = <&gpio0 RK_PA1 GPIO_ACTIVE_HIGH>; // ri
-
-		dtr-gpios = <&gpio0 RK_PA3 GPIO_ACTIVE_HIGH>;
-		cts-gpios = <&gpio3 RK_PC0 GPIO_ACTIVE_HIGH>;
-		rts-gpios = <&gpio3 RK_PC1 GPIO_ACTIVE_HIGH>;
-
-		quectel,qdai = "3,0,0,4,0,0,1,1";
-	};
-};
-
 &pmu_io_domains {
 	pmu1830-supply = <&vcc_1v8>;
 	status = "okay";
@@ -1295,29 +1242,6 @@ flash_pins: flash-pins {
 		};
 	};
 
-	modem {
-		vcc_4g_5v_en: vcc-4g-5v-en-pin {
-			rockchip,pins = <1 RK_PC7 RK_FUNC_GPIO &pcfg_pull_none>;
-		};
-
-		vcc_4g_en: vcc-4g-en-pin {
-			rockchip,pins = <4 RK_PC7 RK_FUNC_GPIO &pcfg_pull_none>;
-		};
-
-		modem_control_pins: modem-control-pins {
-			rockchip,pins =
-				<0 RK_PB0 RK_FUNC_GPIO &pcfg_pull_none>,
-				<3 RK_PB0 RK_FUNC_GPIO &pcfg_pull_none>,
-				<3 RK_PA6 RK_FUNC_GPIO &pcfg_pull_none>,
-				<0 RK_PB5 RK_FUNC_GPIO &pcfg_pull_none>,
-				<0 RK_PB4 RK_FUNC_GPIO &pcfg_pull_none>,
-				<0 RK_PA1 RK_FUNC_GPIO &pcfg_pull_none>,
-				<0 RK_PA3 RK_FUNC_GPIO &pcfg_pull_none>,
-				<3 RK_PC0 RK_FUNC_GPIO &pcfg_pull_none>,
-				<3 RK_PC1 RK_FUNC_GPIO &pcfg_pull_none>;
-		};
-	};
-
 	pmic {
 		pmic_int_l: pmic-int-l {
 			rockchip,pins = <1 RK_PC5 RK_FUNC_GPIO &pcfg_pull_up>;
-- 
2.37.2

